src/tools/dev/scripts/bv_support/bv_adios2.sh

```diff
@@ -133,45 +132,49 @@ function bv_adios2_dry_run
 function build_adios2
 {
     #
-    # ADIOS2 uses CMake  -- make sure we have it built.
-    #
-    CMAKE_INSTALL=${CMAKE_INSTALL:-"$VISITDIR/cmake/${CMAKE_VERSION}/$VISITARCH/bin"}
-    if [[ -e ${CMAKE_INSTALL}/cmake ]] ; then
-        info "ADIOS2: CMake found"
-    else
-        build_cmake
-        if [[ $? != 0 ]] ; then
-            warn "Unable to build cmake.  Giving up"
-            return 1
-        fi
-    fi
-
-    #
-    # Prepare build dir
+    # Uncompress the source file
     #
-    prepare_build_dir $ADIOS2_BUILD_DIR $ADIOS2_FILE
+    uncompress_src_file $ADIOS2_SRC_DIR $ADIOS2_FILE
     untarred_ADIOS2=$?
     if [[ $untarred_ADIOS2 == -1 ]] ; then
-        warn "Unable to prepare ADIOS2 Build Directory. Giving Up"
+        warn "Unable to uncompress ADIOS2 source file. Giving Up!"
         return 1
     fi
+
     #### begin parallel
+    build_types="ser"
 
-    par_build_types="ser"
     if [[ "$parallel" == "yes" ]]; then
-        par_build_types="$par_build_types par"
+        build_types="$build_types par"
     fi
 
-    ADIOS2_SRC_DIR=$ADIOS2_BUILD_DIR
+    for bt in $build_types; do
 
-    for bt in $par_build_types; do
+        ADIOS2_BUILD_DIR="${ADIOS2_SRC_DIR}-build-$bt"
 
-        # Configure.
-        cd $ADIOS2_SRC_DIR || error "Can't cd to $ADIOS2_SRC_DIR"
-        info "Configuring ADIOS2-$bt (~1 minute)"
+        #
```

Comment:
Ok, so here is some out of source work. Other than making the dir and `cd`'ing into and out of it, what is really different about doing business out of source from before?

Also, it looks like the motivation for out of source might be that you need to build this module multiple different ways? Do we have a similar issue with 2D and 3D boxlib or serial/parallel MOAB? If so, how has that been handled there.

Reply:
Nothing functional ... previously the variable BUILD would be used. Then all of sudden there would be a context switch, with a note that the BULD dir is really the SRC dir so SRC = BUILD  and BUILD += "-build". I found that really sloppy.

In the case of moab two dirs are created under the SRC dir. But they could be "out" of the source like ADIOS2.

Box lib looks to create 2d and 3d libs in the dir.